internal/utils/navitia_datetime.go

```diff
@@ -0,0 +1,89 @@
+package utils
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/araddon/dateparse"
+)
+
+func AcceptedDateTime(datetime string) bool {
+	// DateTime format accepted by navitia: https://doc.navitia.io/#standard-objects
+	accetpted_format := []string{
+		// YYYYMMDDThhmmss
+		`(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})$`,
+		// YYYY-MM-DDThh:mm:ss
+		`(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})$`,
+		// YYYY-MM-DDThh:mm:ssZ
+		`(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})(Z)$`,
+		// YYYYMMDDThhmmss+HHMM
+		`(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})([\+,-])(\d{2})(\d{2})$`,
+		// "YYYY-MM-DDThh:mm:ss+HH:MM"
+		`(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})([\+,-])(\d{2}):(\d{2})$`,
+		// "YYYYMMDDThhmmssZ"
+		`(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})(Z)$`,
+	}
+
+	for _, pattern := range accetpted_format {
+		re := regexp.MustCompile(pattern)
+		if re.MatchString(datetime) {
+			return true
+		}
+	}
+	return false
+}
+
+func ParseNavitiaDateTime(datetime string, location string) (time.Time, error) {
+
+	if !AcceptedDateTime(datetime) {
+		return time.Time{}, fmt.Errorf("could not find format for \"%s\"", datetime)
+	}
+
+	loc, _ := time.LoadLocation(location)
+	// datetime with format: YYYYMMDDThhmmss
+	pattern := `(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})$`
+	matched, _ := regexp.MatchString(pattern, datetime)
+	if matched {
+		t, err := time.ParseInLocation("20060102T150405", datetime, loc)
+		if err == nil {
+			return t.UTC(), nil
+		}
+	}
+	// datetime with format: YYYY-MM-DDThh:mm:ss
+	pattern = `(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})$`
+	matched, _ = regexp.MatchString(pattern, datetime)
+	if matched {
+		t, err := time.ParseInLocation("2006-01-02T15:04:05", datetime, loc)
+		if err == nil {
+			return t.UTC(), nil
+		}
+	}
+
+	// datetime with format: YYYYMMDDThhmmssZ
+	pattern = `(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})(Z)$`
+	matched, _ = regexp.MatchString(pattern, datetime)
+	if matched {
+		dtf := regexp.MustCompile(pattern)
+		val := dtf.ReplaceAllString(datetime, "$1$2$3") + "T" + dtf.ReplaceAllString(datetime, "$4$5$6")
+		t, err := time.Parse("20060102T150405", val)
+		if err == nil {
+			return t.UTC(), nil
+		}
+	}
+
+	// datetime with format: YYYYMMDDThhmmss+HHMM --> YYYY-MM-DDThh:mm:ss+HHMM
+	pattern = `(\d{4})(\d{2})(\d{2})T(\d{2})(\d{2})(\d{2})([\+,-])(\d{2})(\d{2})$`
+	matched, _ = regexp.MatchString(pattern, datetime)
+	if matched {
+		dtf := regexp.MustCompile(pattern)
+		val := dtf.ReplaceAllString(datetime, "$1-$2-$3") + "T" + dtf.ReplaceAllString(datetime, "$4:$5:$6$7$8")
```

Comment:
Aren't you forgetting minutes, here? (you may also change the pattern to have 4 digits in TZ)
```suggestion
		val := dtf.ReplaceAllString(datetime, "$1-$2-$3") + "T" + dtf.ReplaceAllString(datetime, "$4:$5:$6$7$8$9")
```

Reply:
:+1: